### PR TITLE
WIP: WCS linking with glue-jupyter

### LINKED ISSUE #25: What does Glue do with WCS and reprojection in Jupyter notebook?
Duplicate of glue-viz/glue-jupyter#112 especially created for JIRA.

p.s. @astrofrog is doing the real work. I am just being assigned in name. 😬

### PR DESCRIPTION
**Pull request by pllim**
Fix #25

### CODE REVIEW

File: examples/glueviz_wcs_example.ipynb [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+    "pip install git+https://github.com/glue-viz/glue-jupyter.git\n",
+    "```\n",
+    "\n",
+    "The versions used to develop this:\n",
+    "\n",
+    "```\n",
+    "# Name                    Version                   Build  Channel\n",
+    "glue-core                 0.15.0.dev5617           pypi_0    pypi\n",
+    "glue-jupyter              0.0.0                    pypi_0    pypi\n",
+    "glue-vispy-viewers        0.12.dev0                pypi_0    pypi\n",
```
Comment (pllim) [inline examples/glueviz_wcs_example.ipynb]: Note: I was also using
```
bqplot                    0.12.0a0                 pypi_0    pypi
bqplot-image-gl           0.1.4                    pypi_0    pypi
```

but I just updated `bqplot` to

```
bqplot                    0.12.0a1                 pypi_0    pypi
```

I'll have to add the version info for `bqplot*` packages next time I update this.

File: examples/glueviz_wcs_example.ipynb [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+    "**TODO: What does this really do? Even with this added, when I pan in first viewer, the second one does not auto-pan.**"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "app.add_link(dither1, 'Right Ascension', dither2, 'Right Ascension')\n",
```
Comment (pllim) [inline examples/glueviz_wcs_example.ipynb]: I thought this was the right way to link the data together. But even when this is executed, panning `viewer1` below does not auto-pan display in `viewer2`.

In contrast, when I select a subset on `viewer1`, the selection box shows up on both viewers and appear to be sync-ing properly.

File: examples/glueviz_wcs_example.ipynb [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# DOES NOT WORK!\n",
+    "# http://glueviz.org/glue-jupyter/developer_notes.html\n",
+    "\n",
+    "#from glue_jupyter.link import link as gjlink\n",
+    "#gjlink((viewer1.state, 'visible'), (viewer2.state, 'visible'))"
```
Comment (pllim) [inline examples/glueviz_wcs_example.ipynb]: http://glueviz.org/glue-jupyter/developer_notes.html has some notes about using bi-directional links with `link(...)` function. But the examples given are some custom viewers that I am not interested in writing. I cannot get this to work. But I am not even sure if this is needed. Won't the data linking above be sufficient? Do I have to link so many different things?

File: examples/glueviz_wcs_example.ipynb (hunk 2 above)
Comment (astrofrog) [reply examples/glueviz_wcs_example.ipynb]: Just to be clear, linking the data means you can show the two datasets in the *same* viewer as overlays to each other (or e.g. as an RGB image if you have three datasets). Linking the views is different - I'll post an example of how to do this.

File: examples/glueviz_wcs_example.ipynb [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+    "We display both dithers in separate, but connected, viewers."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "viewer1 = app.imshow(data=dither1)"
```
[the remark this answers is not in the document]
Comment (pllim) [reply examples/glueviz_wcs_example.ipynb]: So far so good with bqplot 0.12.0a1

File: examples/glueviz_wcs_example.ipynb [hunk 5]
```diff
@@ -0,0 +1,227 @@
...
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Tom R advised not to lock down all four corners with\n",
+    "# auto aspect ratio on or bad things will happen.\n",
+    "keep_in_sync(viewer1.state, 'x_min', viewer2.state, 'x_min')\n",
```
Comment (pllim) [inline examples/glueviz_wcs_example.ipynb]: @astrofrog , still not sure if this is what we want. The goal is to sync by WCS, not pixel location.

**Pull request closed without merging**